### PR TITLE
Add compat entries to experiments/AMIP/modular Project.toml

### LINKED ISSUE #284: Add `compat` entries to `experiments/AMIP/modular/Project.toml`
Tracked as part of SDI #281

### PR DESCRIPTION
**Pull request by valeriabarra**
## Purpose 
The purpose of this PR is to add compat entries to the `experiments/AMIP/modular/Project.toml` so that we can automatically update all dependencies in that environment using the package manager. 

Closes #284 

## Content
Solution implemented:
- Used the package `CompatDevTools` to populate the compat entries
- Removed all patch versions
- For versions > 1.0, I used the same as the `ClimaAtmos.jl/examples` environment (keeping the minors)

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by valeriabarra** [comment]
Everything seems to be passing here. Should I add compat entries also to the top-level `ClimaCoupler.jl/Project.toml` environment in the same way in this PR?

File: experiments/AMIP/modular/Project.toml [hunk 1]
```diff
@@ -48,7 +48,49 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
 UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 
 [compat]
-ClimaTimeSteppers = "0.6.0"
+ArgParse = "1.1"
```
Comment (charleskawczynski) [inline experiments/AMIP/modular/Project.toml:51]:
```suggestion
ArgParse = "1"
```

File: experiments/AMIP/modular/Project.toml [hunk 2]
```diff
@@ -48,7 +48,49 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
...
+ClimaTimeSteppers = "0.6"
+CloudMicrophysics = "0.9"
+Colors = "0.12"
+Dierckx = "0.5"
+DiffEqBase = "6"
+DiffEqCallbacks = "2"
+DocStringExtensions = "0.8, 0.9"
+ForwardDiff = "0.10"
+GR = "0.71"
+Glob = "1.3"
```
Comment (charleskawczynski) [inline experiments/AMIP/modular/Project.toml:72]:
```suggestion
Glob = "1"
```

File: experiments/AMIP/modular/Project.toml [hunk 3]
```diff
@@ -48,7 +48,49 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
...
+HDF5_jll = "1"
+Insolation = "0.3"
+IntervalSets = "0.5, 0.6, 0.7"
+JLD2 = "0.4"
+JSON = "0.21"
+MPI = "0.20"
+NCDatasets = "0.11, 0.12"
+NVTX = "0.3"
+OrdinaryDiffEq = "5, 6"
+Plots = "1.27"
```
Comment (charleskawczynski) [inline experiments/AMIP/modular/Project.toml:82]:
```suggestion
Plots = "1"
```

File: experiments/AMIP/modular/Project.toml [hunk 4]
```diff
@@ -48,7 +48,49 @@ Thermodynamics = "b60c26fb-14c3-4610-9d3e-2d17fe7ff00c"
...
+PrettyTables = "1.3, 2"
```
Comment (charleskawczynski) [inline experiments/AMIP/modular/Project.toml:83]:
```suggestion
PrettyTables = "1, 2"
```

**Review by charleskawczynski — APPROVED** [review]

File: experiments/AMIP/modular/Project.toml (hunk 1 above)
Comment (valeriabarra) [reply experiments/AMIP/modular/Project.toml:51]: Hi @charleskawczynski , thanks. As I mentioned in the PR description, or versions > 1.0, I used the same as the ClimaAtmos.jl/examples environment (keeping the minors), as a reference and to make sure they are compatible. See [`examples/Project.toml`](https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/Project.toml). What should we do?

Comment (charleskawczynski) [reply experiments/AMIP/modular/Project.toml:51]: I'm updating ClimaAtmos' compat entries in https://github.com/CliMA/ClimaAtmos.jl/pull/1603, so let's just apply my suggestion

**Review by LenkaNovak — COMMENTED** [review]
Looks good. But can you guys specify what should(n't) do in the [compact] list? E.g. why not Test, Statistics, Random... etc?

**Comment by charleskawczynski** [comment]
> Looks good. But can you guys specify what should(n't) go in the [compact] list? E.g. why not Test, Statistics, Random... etc?

All dependencies with versions should go into the compat list-- Test, Statistics etc. are in the stdlib, which versions are tied to the julia version

**Comment by valeriabarra** [comment]
> > Everything seems to be passing here. Should I add compat entries also to the top-level `ClimaCoupler.jl/Project.toml` environment in the same way in this PR?
> 
> Yeah, that'd be a good idea I think. `CompatDevTools` does have a utility for synchronizing compat entries, but it by default scans all directories, so it may not be the easiest to use for syncing them.

@charleskawczynski , I added them for all Projects: top-level, `experiments/AMIP/modular`, `test`, and `perf`.

**Comment by LenkaNovak** [comment]
> > Looks good. But can you guys specify what should(n't) go in the [compact] list? E.g. why not Test, Statistics, Random... etc?
> 
> All dependencies with versions should go into the compat list-- Test, Statistics etc. are in the stdlib, which versions are tied to the julia version

Great, thank you for clarifying!

**Pull request merged**